app/app.js

```diff
@@ -13,10 +13,15 @@ const v1Router = require('./src/routes/v1');
 const DataConnection = require('./src/db/dataConnection');
 const dataConnection = new DataConnection();
 
+const realmAdminInit = require('./src/components/realmAdminService').initialize;
+const realmAdminService = require('./src/components/realmAdminService').realmAdminService;
+const userSyncService = require('./src/forms/form/userSyncService');
```

Comment:
I'm wondering if we need to be putting these services into top level app. We're not really keeping a persistent TCP connection to Keycloak, or at least I don't envision us having a need to do so, as we'd only need that information when someone eventually views the admin page to see incoming requests. I think we should only be reserving things on app level as a "connection" if we must 1) kill the app if said "connection" is unreachable or 2) we need more stringent monitoring of an external resource that immediately impacts the ability for the backend service to function.

Reply:
I'm going to kill this PR because it's easier to do that than do a massive refactor.  The reason this is at the app level is it kicks off a user sync on start up and a polling that will continually grab new realm users and put them in the db (only kc id and email).  How/where would you trigger such things and start polling jobs?

Reply:
We need an ability for a user after they login, to be able to request for some form of access. That would effectively be a "push" that we can record into the DB at the time of request (they're logged in, so we can get their kc id and email from the jwt).

I'm not convinced that we need a polling "pull" model because for the incoming admin interface, our aim is to just list a finite set of unhandled requests - which can be fully populated from all the "push" requests events that happened. When a request has been acted upon, we can just drop that record from the db (as for now we've scoped out the notion of needing to retain action history as per sprint planning chat). We don't "really" have to poll keycloak constantly because we can easily just make an accept request click on the panel apply the requested role/group (even if they already have it, so should resolve to no-action), or reject request (no action anyways).

Reply:
Yeah just finally getting to look at this now, but do we need user syncing, or even recording users at all in the system?
Would this flow work (I think this is the UX we've just discussed):
1 - User logs in, they're entered in KC
2 - User clicks request access, this writes an access request record for the app type with KC guid.
3 - Admin goes to admin panel and sees pending access requests from that table
4 - Admin approves, this sends a group add with the associated KC guid to KC
5 - Access request row is marked as done or deleted or whatever

Reply:
alright, i can take polling out for sure.  just thought it'd help with searches to add users in app - could be a lot quicker than hitting KC.  But it can stay out until that day comes.